### PR TITLE
Daffy merge

### PR DESCRIPTION
**Pull request by afdaniele**

### CODE REVIEW

File: code/autolab/logging_utils.py [hunk 1]
```diff
@@ -2,27 +2,30 @@
...
-        docker.containers.get(name).remove(force=True)
+        docker.containers.get(BAG_RECORDER).remove(force=True)
     except:
         pass
 
     # Create the log folder
-    print(mount_folder)
-    os.makedirs(mount_folder, exist_ok=True)
+    if not os.path.exists(mount_folder):
+        os.makedirs(mount_folder)
```
Comment (AmaurX) [inline code/autolab/logging_utils.py]: @afdaniele  isnt the (exist_ok = true ) option better ?

File: code/autolab/logging_utils.py [hunk 2]
```diff
@@ -34,42 +37,41 @@ def start_logging(filename, device_list, mount_folder):
...
-
-
-# if __name__ == "__main__":
-#     start_logging("test", ["autobot02"], "/home/amaury")
-#     time.sleep(5)
-#     stop_logging()
+    # stop recorder and container (if running)
+    if container.status == 'running':
+        print("Stopping bag recorder")
+        cmd = "/bin/bash -c 'rosnode kill bag_recorder_node'"
```
Comment (afdaniele) [inline code/autolab/logging_utils.py]: this should go!

File: dependencies-py3.txt [hunk 3]
```diff
@@ -6,3 +6,7 @@ boto3
 pyaml
 docker
 duckietown-shell
+
+
+
+
```
Comment (afdaniele) [inline dependencies-py3.txt]: these should go

File: docker-compose.yaml [hunk 4]
```diff
@@ -10,8 +10,8 @@ services:
     volumes:
       - ./code/autolab:/code/dt-autolab-api/autolab
       - /var/run/docker.sock:/var/run/docker.sock
-      # - /tmp/aido_ws/:/tmp/aido_ws/
-      - ~/.ssh/:/external_ssh
+      - /tmp/aido_ws/:/tmp/aido_ws/
```
Comment (afdaniele) [inline docker-compose.yaml]: this should go!

Comment (AmaurX) [inline docker-compose.yaml]: not needed thanks to /tmp:/tmp

File: reset_duckiebot.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import subprocess
```
Comment (AmaurX) [inline reset_duckiebot.py]: @afdaniele this whole file is not at the right place

**Review by AmaurX — CHANGES_REQUESTED** [review]

File: reset_duckiebot.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    results = pool.map(start_device, device_list)
+    pool.close()
+    pool.join()
+    show_status(device_list, results)
+    return results
+
+
+def reset_duckiebot_with_list(device_list):
+    outcome = start_all_devices(device_list)
+    return device_list, outcome
```
Comment (afdaniele) [inline reset_duckiebot.py]: this file should go entirely!

File: code/autolab/logging_utils.py (hunk 1 above)
Comment (AmaurX) [reply code/autolab/logging_utils.py]: but the print statement should indeed go

Comment (afdaniele) [reply code/autolab/logging_utils.py]: @AmaurX 
Let's try with exist_ok

**Review by afdaniele — COMMENTED** [review]
fixed!

File: code/autolab/logging_utils.py (hunk 2 above)
Comment (afdaniele) [reply code/autolab/logging_utils.py]: done

File: docker-compose.yaml (hunk 4 above)
Comment (afdaniele) [reply docker-compose.yaml]: done

**Pull request merged**